File: app/src/main/java/com/ndmsystems/coala/CoAPHandler.java
package com.ndmsystems.coala;


import com.ndmsystems.coala.exceptions.BaseCoalaThrowable;
import com.ndmsystems.coala.message.CoAPMessage;

public interface CoAPHandler {
    void onMessage(CoAPMessage message, String error);

    void onAckError(String error);

    class AckError extends BaseCoalaThrowable {
        public AckError() {
        }

        public AckError(String message) {
            super(message);
        }
    }
}

File: app/src/main/java/com/ndmsystems/coala/CoAPHandler.kt
package com.ndmsystems.coala

import com.ndmsystems.coala.exceptions.BaseCoalaThrowable
import com.ndmsystems.coala.message.CoAPMessage

interface CoAPHandler {
    fun onMessage(message: CoAPMessage, error: String?)
    fun onAckError(error: String)
    class AckError(message: String) : BaseCoalaThrowable(message)
}